File: scripts/dump_schema.js
import Database from 'better-sqlite3';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const dbPath = path.join(__dirname, '../autox.db');
const db = new Database(dbPath);

const tables = db.prepare("SELECT name, sql FROM sqlite_master WHERE type='table'").all();
console.log(JSON.stringify(tables, null, 2));

db.close();

File: dump_schema.py
import sqlite3

db_path = 'autox.db'
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

output_file = 'schema_info.txt'

with open(output_file, 'w') as f:
    cursor.execute("SELECT sql FROM sqlite_master WHERE type='table';")
    rows = cursor.fetchall()
    for row in rows:
        f.write(row[0] + ";\n\n")

conn.close()
